app/modules/encounters/resources.py

```diff
@@ -104,9 +165,123 @@ def delete(self, encounter):
         """
         Delete a Encounter by ID.
         """
+
+        # first try delete on edm
+        target = 'default'
+        path = str(encounter.guid)
+        request_func = current_app.edm.delete_passthrough
+        passthrough_kwargs = {}
+        response = _request_passthrough(target, path, request_func, passthrough_kwargs)
+
+        response_data = None
+        result_data = None
+        if response.ok:
+            response_data = response.json()
+            result_data = response_data.get('result', None)
+
+        if (
+            not response.ok
+            or not response_data.get('success', False)
+            or result_data is None
+        ):
+            log.warning('Encounter.delete %r failed' % (encounter.guid))
+            passed_message = {'message': {'key': 'error'}}
+            if response_data is not None and 'message' in response_data:
+                passed_message = response_data['message']
+            abort(success=False, passed_message=passed_message, message='Error', code=400)
+
+        # if we get here, edm has deleted the encounter, now houston feather
         context = api.commit_or_abort(
             db.session, default_error_message='Failed to delete the Encounter.'
         )
+        # TODO handle failure of feather deletion (when edm successful!)  out-of-sync == bad
         with context:
             db.session.delete(encounter)
         return None
+
+
+@api.route('/<uuid:encounter_guid>/complete')
+@api.login_required(oauth_scopes=['encounters:read'])
+@api.response(
+    code=HTTPStatus.NOT_FOUND,
+    description='Encounter not found.',
+)
+@api.resolve_object_by_model(Encounter, 'encounter', return_not_found=True)
+class EncounterByIDComplete(Resource):
+
+    # @api.response(schemas.DetailedEncounterSchema())
+    def get(self, encounter):
+        """
+        Get Encounter full details by ID.
+        """
+        encounter, encounter_guids = encounter
+
+        if encounter is not None:
+            print('####### found encounter within houston : %s' % (encounter,))
+            # note: should probably _still_ check edm for: stale cache, deletion!
+            #      user.edm_sync(version)
+            # return encounter
+            # return True
+
+        response = current_app.edm.get_encounter_data_dict(encounter.guid)
+        # TODO handle non-200 ?
+        # assert response.success
+        # import utool as ut
+        # ut.embed()
+        return response['result']
+
+
+def _request_passthrough(target, path, request_func, passthrough_kwargs):
```

Comment:
A question rather than an instruction.
Is passthrough a thing that is encounter only or will it be for many things? I thought it was the latter.
If so then the only thing encounter specific in this method is the Encounter bit in the endpoint.
Could that be a parameter and this function moved entirely to the edm extension?
Otherwise I see a lot of copy & bodge in this area.

Reply:
yes... soon enough i suspect this is exactly what will happen.  mostly wanted to get this _functional_ in the encounter world.  in fact, this has _already_ been sloppily copied (by me!) from [its original location](https://github.com/WildMeOrg/houston/blob/master/app/modules/passthroughs/resources.py#L44).   so i definitely need to generalize and consolidate this ....... all over the place.   :smile: